client/state/data-layer/wpcom-http/actions.js

```diff
@@ -47,7 +47,7 @@ export const http = ( {
 		body,
 		method,
 		path,
-		query: method === 'GET' ? { ...query, ...version } : version,
+		query: method === 'GET' && { ...query, version },
```

Comment:
Also, query params are perfectly valid on non GET requests. We shouldn't be restricting them in the first place.